app/javascript/packs/followButtons.js

```diff
@@ -409,33 +411,38 @@ function initializeNonUserFollowButtons() {
     '.follow-action-button:not(.follow-user):not([data-fetched])',
   );
 
-  const userLoggedIn =
-    document.body.getAttribute('data-user-status') === 'logged-in';
-
-  const user = userLoggedIn ? userData() : null;
-
-  const followedTags = user
-    ? JSON.parse(user.followed_tags).map((tag) => tag.id)
-    : [];
-
-  const followedTagIds = new Set(followedTags);
-
-  nonUserFollowButtons.forEach((button) => {
-    const { info } = button.dataset;
-    const buttonInfo = JSON.parse(info);
-    const { className, name } = buttonInfo;
-    addAriaLabelToButton({ button, followType: className, followName: name });
-    if (className === 'Tag' && user) {
-      // We don't need to make a network request to 'fetch' the status of tag buttons
-      button.dataset.fetched = true;
-      const initialButtonFollowState = followedTagIds.has(buttonInfo.id)
-        ? 'true'
-        : 'false';
-      updateInitialButtonUI(initialButtonFollowState, button);
-    } else {
-      fetchFollowButtonStatus(button, buttonInfo);
-    }
-  });
+  waitOnBaseData()
```

Comment:
So, we also have an odd race-condition type of thing here, where the JavaScript that initializes the `follow button` might finish before the `/async_info/base_data` finishes. When that happens (which, locally, was almost every time), stale data is pulled out of our [browser store cache](https://github.com/forem/forem/blob/main/app/assets/javascripts/utilities/browserStoreCache.js), _before_ that caches is swept by the `async_info` process. A refresh will 'fix' things, but we shouldn't have to refresh twice every time we change our followings. We've had [`waitOnBaseData` for 2+ years](https://github.com/forem/forem/blame/main/app/javascript/utilities/waitOnBaseData.js), but for some reason aren't using it widely. One downside here: we _will_ see a _slightly_ longer delay where _all_ the buttons look like "not following" state, but I don't see a way to fix that without a massive refactor.